src/storage/store/node_table_data.cpp

```diff
@@ -58,28 +58,30 @@ void NodeTableData::initializeScanState(Transaction* transaction, TableScanState
     if (transaction->isWriteTransaction()) {
         initializeLocalNodeReadState(transaction, scanState, scanState.nodeGroupIdx);
     }
-    dataScanState.vectorIdx = INVALID_VECTOR_IDX;
+    dataScanState.vectorIdx = INVALID_IDX;
     dataScanState.numRowsInNodeGroup =
         columns[0]->getMetadata(scanState.nodeGroupIdx, TransactionType::READ_ONLY).numValues;
 }
 
 void NodeTableData::initializeColumnScanStates(Transaction* transaction, TableScanState& scanState,
     node_group_idx_t nodeGroupIdx) const {
-    auto& tableScanState = scanState.cast<NodeTableScanState>();
-    auto& dataReadState = scanState.dataScanState->cast<NodeDataScanState>();
+    scanState.zoneMapResult = ZoneMapCheckResult::ALWAYS_SCAN;
+    auto& dataScanState = scanState.dataScanState->cast<NodeDataScanState>();
     for (auto i = 0u; i < scanState.columnIDs.size(); i++) {
```

Comment:
I think we should break this for loop into two. The first one is to simply gather all scanStates. The second one is when `!scanState.columnPredicateSets.empty()` and check zone map to set `scanState.zoneMapResult`.